src/tcgwars/logic/groovy/TcgStatics.groovy

```diff
@@ -1146,6 +1146,72 @@ class TcgStatics {
   static boolean isGxPerformed(){
     bg.em().retrieveObject("gx_"+my.owner)
   }
+
+  /* General checks for attacks and abilities */
+
+  //  * assertBench
+  //    - Can be used with no arguments, or with a condition and additional text included.
+  //    - If a closure is given, it'll check for any "it" element in bench returning true.
+  //    - Optional params:
+  //      + type: if set, restricts to benched Pokémon of a single specific type.
+  //      + isEX/isGX/isV: Can be expanded if needed. All of these unset will have the method search for any Pokémon no matter what, but if even a single one is set true it'll only filter those that are set true as well.
+  //      + opp: If true, checks for the opponent's bench instead of "my" bench.
+  //      + repText: If true, instead of adding cText at the end of the assert it'll be the only thing printed.
+  static void assertBench(params=[:], Closure c, String cText) {
+    def checkedBench = params.opp ? opp.bench : my.bench
+
+    def isPokeCnt = 0
+    [params.isEX, params.isGX, params.isV].each{isPokeCnt += 1}
+
+    def benchFilter = (
+      if (c == null) { true } else {
+        (
+          if (params.type) {it.types.contains(params.type)} else {true}
+        ) && (
+          if (!isPokeCnt) {true} else {
+            (params.isEX && it.pokemonEX) ||
+            (params.isGX && it.pokemonGX) ||
+            (params.isV && it.pokemonV)
+          }
+        ) && (
+          c(it)
+        )
+      }
+    )
+
+    def pokeString = "Pokémon"
+    if (isPokeCnt) (
+      pokeString = ""
+      i = 1
+      [(params.isEX, "Pokémon-EX"), (params.isGX, "Pokémon-GX"), (params.isV, "Pokémon V")].each{
+        if (it[0]) {
+          pokeString += it[1] + (if (i == isPokeCnt) "" else if (i == isPokeCnt-1) " or " else ", ")
+          i += 1
+        }
+      }
+    )
+
+    def failMessage = if (params.repText) { cText } else {
+      "You" + (params.opp ? "r opponent does" : " do") + "n't have any " + (
+        param.type ? params.type.getShortNotation() + " " : ""
+      ) + "Benched $pokeString" + () + (
+        if (c != null) {
+          " that " + (cText ? cText : "follow the stated condition(s)")
+        }
+      )
+    }
+
+    assert benchFilter : failMessage
+  }
+  static void assertMyBench(params=[:], Closure c, String cText) {
+    params.opp = false
+    assertBench(params, c, cText)
+  }
+  static void assertOppBench(params=[:], Closure c, String cText) {
+    params.opp = true
+    assertBench(params, c, cText)
+  }
```

Comment:
I don't understand why cText is a required parameter here

Reply:
It's not, though the check is done for null inside the method. Could probably use `.equals("")` though.

Reply:
In this case the method won't run without a default value assignment. Change the parameter to `String cText = null`